### DISCUSSION TITLE
Is it safe to include prisma client in a front end react app?

### DISCUSSION THREAD

**Discussion by howagain** [prisma/prisma#9397] [Q&A]
I assume not since you have direct DB access, but the fact that it's called Prisma "Client" makes me wonder if you can use the prisma client on front end code delivered to the end user and it creates an API without leaking db secrets or posing other security risks, akin to blitz or supabase.

**Comment by ryands17** [comment]
@howagain 👋 

At the moment, Prisma cannot be used in a front end react app. The term [Prisma Client](https://www.prisma.io/docs/concepts/components/prisma-client) only refers to a part of Prisma that helps you write type-safe queries and has got nothing to do with the frontend.

As for Blitz, it's not a fully frontend app and uses the server part of Next.js to run queries (Blitz uses Prisma btw).

**Comment by howagain** [comment]
Thanks for clearing that up :) Might want to put that in big warning
letters on the client app documentation so people don’t make stupid
mistakes and try to use it like firebase or some other client side sdk and
upload their db secret to the world.

**Reply by MentalGear** [reply]
I agree, "client" is definitely not the best word chosen. Should be more something like "Prisma Interface" or so.

**Comment by TimotiusAnrez** [comment]
Can we use getPayload api to get type from prisma to be used in client side application? it is really useful to have that kind of flexible type checking instead of shouldering the burden of maintaining dozens of types that we need to anticipated for our components?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
